### PR TITLE
Properly parse "Fixes:"

### PR DESCRIPTION
**Pull request by tniessen**
Fixes: https://github.com/joyeecheung/node-core-utils/issues/44

### CODE REVIEW

File: test/unit/links.test.js [hunk 1]
```diff
@@ -6,18 +6,21 @@ const assert = require('assert');
 const htmls = fixtures.readJSON('op_html.json');
 
 const expected = [{
-  fixes: ['https://github.com/node/issues/16437'],
+  fixes: ['https://github.com/nodejs/node/issues/16437'],
```
Comment (tniessen) [inline test/unit/links.test.js]: I believe this test was incorrect before. Not sure whether I am supposed to pass `nodejs/node` as the repo name.

**Comment by apapirovski** [comment]
This seems great! Wondering two things: a) should this be case-insensitive matching and b) should we enforce the colon? "fixes #XYZ" seems valid to me...

**Comment by tniessen** [comment]
@apapirovski Good points, I think both should be changed!

File: test/unit/links.test.js (hunk 1 above)
Comment (joyeecheung) [reply test/unit/links.test.js]: I think we can just pass the repo + the owner name to `LinkParser`

Comment (joyeecheung) [reply test/unit/links.test.js]: ^to be clear, I meant change the signature of  the constructor of `LinkParser` to `(owner, repo, html)`

**Comment by tniessen** [comment]
@apapirovski I made the checks case-insensitive. Making the colon optional requires much stricter parsing as the keywords can appear in regular sentences as well.

**Comment by addaleax** [comment]
@tniessen If I’m reading this correctly the regexes still expect the colon to always be there, right?

**Comment by apapirovski** [comment]
I feel like it's probably fine if we parse "fixes #XYZ" anywhere in the copy, no? It's possible I'm missing some downside to that approach though.

**Comment by tniessen** [comment]
> If I’m reading this correctly the regexes still expect the colon to always be there, right?

@addaleax Yes, making the colon optional requires stricter filtering. I am not sure whether we can assume that any Fixes-like keyword followed by `#(decimal)` or by `(word)/(word)/(decimal)` is valid and everything else is not, but it is probably enough.

**Comment by apapirovski** [comment]
We could get fancy and check the API to make sure it's an actual issue 😆 I know, I know...

It seems fine to me as is and we can revisit later, maybe including validating that these are actual open issues.

**Comment by joyeecheung** [comment]
BTW github should have some kind of API about the status of issues they consider can be automatically closed(https://help.github.com/articles/closing-issues-using-keywords/), but it doesn't seem very stable at this point.

**Comment by joyeecheung** [comment]
@tniessen I think this can be merged now?

**Comment by tniessen** [comment]
I will revisit the version without a colon later.

**Pull request merged**